Perception/street_sign_detector/src/street_sign_detector/street_sign_detector.py

```diff
@@ -0,0 +1,272 @@
+import sys
+import traceback
+
+darknet_path = str(pathlib.Path(__file__).parent.parent.parent.absolute() / 'darknet')
+print(darknet_path)
+sys.path.append(darknet_path)
+from darknet_carla import YOLO
+from darknet import print_detections
+
+
+class StreetSignDetector:
+    """
+    Class to use /carla/<role_name>/semantic_segmentation_front/image and /carla/<role_name>/rgb_front/image topic
+    to detect street signs and publish them to /psaf/<role_name>/street_sign topic.
+    """
+
+    def __init__(self, role_name):
+        self.role_name = role_name
+        self.bridge = CvBridge()
+        self._semantic_segmentation_front_image_subscriber = rospy.Subscriber(
+            "/carla/{}/semantic_segmentation_front/image".format(role_name), Image,
+            self.semantic_segmentation_image_updated)
+
+        self._rgb_front_image_subscriber = rospy.Subscriber(
+            "/carla/{}/rgb_front/image".format(role_name), Image, self.rgb_image_updated)
+            
+        self.detection = None
+        self.speed_limit_publisher = rospy.Publisher(
+            "/psaf/{}/speed_limit".format(self.role_name), Float64, queue_size=1, latch=True)
+        self.perception_info_publisher = rospy.Publisher(
+            "/psaf/{}/perception_info".format(self.role_name), PerceptionInfo, queue_size=1, latch=True)
+
+        self._control = CarlaEgoVehicleControl()
+        self.rgb_img = np.zeros((600, 800, 3), np.uint8)
+        self.semantic_segmentation_img = np.zeros((300, 400, 3), np.uint8)
+        self.yolo_detector = YOLO()
+
+        # FOR DEBUG ONLY
+        self.global_filename_counter = 0
+
+    def get_block_by_pixel(self, img_rgb, img_sem, h_current, w_current, height, width, filename):
+        '''
+        Finds the street sign in the image of the segmentation camera which includes the pixel at position (h_current,
+        w_current), cuts the shape out of the rgb camera image and saves the street sign in a file.
+
+        :param img_rgb: current image of the rgb camera
+        :param img_sem: current image of the semantic segmentation camera
+        :param h_current: x position of the currently observed pixel
+        :param w_current: y position of the currently observed pixel
+        :param height: height of img_sem_seg
+        :param width: width of img_sem_seg
+        :param filename: filename for the next street sign
+        :return: the img_sem_seg in which the latest found street sign is removed
+        '''
+        pixels = []
+        current_pixels = []
+        pixels.append([h_current, w_current])
+        current_pixels.append([h_current, w_current])
+        img_sem[h_current, w_current, 2] = 0
+        while len(current_pixels) > 0:
+            for pixel in current_pixels:
+                # check if pixel above is part of the street sign
+                if pixel[0] > 0 and img_sem[pixel[0] - 1, pixel[1], 0] == 0 and img_sem[pixel[0] - 1, pixel[1], 1] == 220 and img_sem[pixel[0] - 1, pixel[1], 2] == 220:
```

Comment:
Ich würde sogar vorschlagen die Farbe noch als Variable festzulegen um magic numbers zu vermeiden
```suggestion
                if pixel[0] > 0 and (img_sem[pixel[0] - 1, pixel[1]] == [0,220,220]).all():
```

Reply:
variablen eingeführt und die ganze logik wird gerade insgesamt überdacht